### ISSUE TITLE
Running when-changed on a VirtualBox guest and watching a file in a shared directory, changes made by the host are not detected

### ISSUE THREAD

**Issue by mh5** [joh/when-changed#40]
Some probably relevant info

    $ pip show when-changed
    ---
    Name: when-changed
    Version: 0.3.0
    Location: /usr/local/lib/python2.7/dist-packages
    Requires: watchdog

Both the host  and guest OSes are debian Jessie.

**Comment by dtgriscom** [comment]
I believe this is an issue with the VirtualBox shared file system, which doesn't properly issue file changed events in the guest when a host file is changed.

**Comment by joh** [comment]
Yeah, not much we can do if we don't get any notifications from the file system. I guess we could fall back to polling...

**Comment by artcz** [comment]
Just run into the same problem, can confirm that doesn't work if file is changed from host, works fine when file is changed from guest. (`while inotifywait -e close_write foo.py; do ./bar.py; done` doesn't work in either case)

+1 for optional polling
